src/main/java/core/basesyntax/service/FruitTransactionServiceImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitTransactionServiceImpl implements FruitTransactionService {
+    private Map<String, Integer> fruitStorage = new HashMap<>();
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        fruitStorage.put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
The `handle` method currently overwrites the fruit quantity in the storage with the transaction's quantity. Consider implementing logic to handle different transaction operations (e.g., BALANCE, SUPPLY, PURCHASE, RETURN) to modify the storage correctly.